src/crypto.rs

```diff
@@ -0,0 +1,97 @@
+use std::fmt::{Debug, Display};
+
+type DynError = Box<dyn std::error::Error + Send + Sync>;
+
+#[derive(Debug)]
+pub struct Error {
```

Comment:
I think it'd be better to just use object_store::Error, whilst it contains variants that aren't relevant, I don't think this matters